local/observeDeviceResources.go

```diff
@@ -198,7 +201,13 @@ func (c *Client) stopObservingDeviceResources(observationID string) (sync func()
 	return sub.Wait, nil
 }
 
-func (c *Client) ObserveDeviceResources(ctx context.Context, deviceID string, handler DeviceResourcesObservationHandler) (string, error) {
+func (c *Client) ObserveDeviceResources(ctx context.Context, deviceID string, handler DeviceResourcesObservationHandler, opts ...CommonCommandOption) (string, error) {
+	cfg := commonCommandOptions{
+		discoveryConfiguration: core.DefaultDiscoveryConfiguration(),
+	}
+	for _, o := range opts {
+		cfg = o.applyOnCommonCommand(cfg)
+	}
```

Comment:
these lines of code are copied in at least 5 places in this MR, consider making into a function

Reply:
fixed